File: src/mastodon_client.py
from mastodon import Mastodon

def init_mastodon(api_base_url, access_token):
    return Mastodon(access_token=access_token, api_base_url=api_base_url)

def get_account_id(api, username):
    try:
        account = api.account_search(username)[0]
        return account['id']
    except Exception as e:
        print(f"Error retrieving account ID for {username}: {e}")
        return None

def fetch_account_posts(api, account_id, clean_func):
    try:
        posts = []
        max_id = None
        while True:
            batch = api.account_statuses(account_id, max_id=max_id, exclude_reblogs=True, exclude_replies=False)
            if not batch:
                break
            posts.extend([clean_func(status["content"]) for status in batch])
            max_id = batch[-1]["id"]
        return posts
    except Exception as e:
        print(f"Error fetching posts: {e}")
        return []

File: src/mastodon_client.rs
//! Mastodon API interaction — auth, fetching, and posting.
//!
//! Thin wrappers over the `mammut` client library. The source
//! account provides the corpus; the destination account publishes.

use mammut::{Data, Mastodon};
use std::borrow::Cow;

/// Build a `Mastodon` client from base URL and access token.
///
/// Only `base` and `token` are required for token-authenticated access;
/// the other `Data` fields are left empty.
pub fn init_mastodon(base_url: &str, access_token: &str) -> Mastodon {
    Mastodon::from_data(Data {
        base: Cow::Owned(base_url.to_string()),
        client_id: Cow::Borrowed(""),
        client_secret: Cow::Borrowed(""),
        redirect: Cow::Borrowed(""),
        token: Cow::Owned(access_token.to_string()),
    })
}

/// Resolve a Mastodon account ID from a username.
///
/// Placeholder — the actual API lookup isn't wired yet.
pub fn get_account_id(_client: &Mastodon, _username: &str) -> Option<String> {
    None // Placeholder
}

/// Fetch all recent posts from an account, cleaning each one.
///
/// Placeholder — the paginated API call isn't wired yet.
/// `cleaner` is a function that strips/normalises each post's text.
pub fn fetch_account_posts<F>(
    _client: &Mastodon,
    _account_id: &str,
    _cleaner: F,
) -> Vec<String>
where
    F: Fn(&str) -> String,
{
    Vec::new() // Placeholder
}
